src/main/java/com/amihaiemil/zold/RtWallet.java

```diff
@@ -143,12 +144,12 @@ public void pay(
     }
 
     @Override
-    public JsonArray find(
-            final String id, final String details
-    ) throws IOException, URISyntaxException {
+    public JsonArray find(final JsonObject params)
+            throws IOException, URISyntaxException {
         URIBuilder builder = new URIBuilder(this.baseUri.toString() + "/find");
-        builder.setParameter("bnf", id);
-        builder.setParameter("details", details);
+        for (final String key: params.keySet()) {
```

Comment:
@ammaratef45 I'd add a hash-set with valid parameter keys and add only valid parameters to URI query